DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/Purity.scala

```diff
@@ -77,9 +73,139 @@ import org.opalj.tac.fpcf.analyses.purity.L1PurityAnalysis
+    private val jdkCommand = getPlainScallopOption(JDKCommand)
+    private val individualCommand = getPlainScallopOption(IndividualCommand)
+    private val closedWorldCommand = getPlainScallopOption(CloseWorldCommand)
+    private val libraryCommand = getPlainScallopOption(LibraryCommand)
+    private val debugCommand = getPlainScallopOption(DebugCommand)
+    private val multiProjectsCommand = getPlainScallopOption(MultiProjectsCommand)
+    private val evaluationDirCommand = getPlainScallopOption(EvalDirCommand)
+    private val packagesCommand = getPlainScallopOption(PackagesCommand)
+    private val threadsNumCommand = getPlainScallopOption(ThreadsNumCommand)
+    private val analysisNameCommand = getPlainScallopOption(AnalysisNameCommand)
+    private val schedulingStrategyCommand = getPlainScallopOption(SchedulingStrategyCommand)
+
+    verify()
+
+    // Parsed data
+    var classPathFiles = parseCommandWithExternalParser(classPathCommand, ClassPathCommandExternalParser).getOrElse(null)
+    var projectDirectory = parseCommandWithInternalParser(projectDirCommand, ProjectDirectoryCommand).getOrElse(null)
+    var libraryDirectory = parseCommandWithInternalParser(libDirCommand, LibraryDirectoryCommand).getOrElse(null)
+    var analysisScheduler = parseCommandWithExternalParser(analysisCommand, AnalysisCommandExternalParser).getOrElse(null)
+    var support: Option[List[FPCFAnalysisScheduler]] = None
+
+    if (fieldAssignabilityCommand.isDefined && escapeCommand.isDefined && eagerCommand.isDefined && analysisScheduler != null) support = Some(parseArgumentsForSupport(
+            analysisCommand.apply(),
+            fieldAssignabilityCommand.apply(),
+            escapeCommand.apply(),
+            eagerCommand.apply(),
+            analysisScheduler
+        )) else support = None
+
+    var domain = parseCommandWithExternalParser(domainCommand, DomainCommandExternalParser).getOrElse(null)
+    var rater = parseCommandWithExternalParser(raterCommand, RaterCommandExternalParser).getOrElse(null)
+    var callGraph = parseCommandWithExternalParser(callGraphCommand, CallGraphCommandExternalParser).getOrElse(null)
+    var jdk: Boolean = parseCommand(jdkCommand).getOrElse(false)
+    var individual: Boolean = parseCommand(individualCommand).getOrElse(false)
+    var closedWorld:Boolean = parseCommand(closedWorldCommand).getOrElse(false)
+    var library: Boolean = parseCommand(libraryCommand).getOrElse(false)
+    var debug: Boolean = parseCommand(debugCommand).getOrElse(false)
+    var multiProjects: Boolean = parseCommand(multiProjectsCommand).getOrElse(false)
+    var evaluationDir = parseCommandWithInternalParser(evaluationDirCommand, EvalDirCommand).getOrElse(null)
+    var packages = parseCommandWithInternalParser(packagesCommand, PackagesCommand).getOrElse(null)
+    var threadsNum: Int = parseCommand(threadsNumCommand).getOrElse(0)
+    var configurationName = parseCommand(analysisNameCommand).getOrElse(null)
```

Comment:
Using `.getOrElse` here is especially strange since you end up wrapping it into an Option again below.